src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx

```diff
@@ -0,0 +1,160 @@
+                                    control={control}
+                                    placeholder="Enter a remote folder name"
+                                    type="text"
+                                    autoComplete="off"
+                                />
+                            </div>
+                            <div className="mb-2">
+                                <Label>Region</Label>
+                                <FormSelectCreatable
+                                    name={getName("awsRegionName")}
+                                    control={control}
+                                    placeholder="Select an AWS region"
+                                    options={availableGlacierRegions}
+                                />
+                            </div>
+                            <div className="mb-2">
+                                <Label>Access key</Label>
+                                <FormInput
+                                    name={getName("awsAccessKey")}
+                                    control={control}
+                                    placeholder="Enter an access key"
+                                    type="text"
+                                    autoComplete="off"
+                                />
+                            </div>
+                            <div className="mb-2">
+                                <Label>Secret key</Label>
+                                <FormInput
+                                    name={getName("awsSecretKey")}
+                                    control={control}
+                                    placeholder="Enter a secret key"
+                                    type="text"
+                                    autoComplete="off"
+                                />
+                            </div>
+                            <div className="d-flex justify-content-end">
+                                <FlexGrow />
+                                <ButtonWithSpinner
+                                    type="button"
+                                    color="secondary"
+                                    onClick={asyncTest.execute}
+                                    isSpinning={asyncTest.loading}
+                                >
+                                    <Icon icon="rocket" />
+                                    Test credentials
+                                </ButtonWithSpinner>
+                            </div>
+                            {asyncTest.result?.Error && (
+                                <div className="mt-3">
+                                    <ConnectionTestResult testResult={asyncTest.result} />
+                                </div>
+                            )}
+                        </div>
+                    )}
+                </Collapse>
+            </CardBody>
+        </Card>
+    );
+}
+
+const fieldBase = "destinations.glacier";
```

Comment:
should we pass this from external ?

Reply:
Maybe, but it doesn't really matter. If I made a typo here, there would be an error in every use of getName().

Reply:
additionally, it is directly linked to this component and never changes

Reply:
if we want to use it in another place, I assume that we must have the `destinations` field there too